samples/nrf9160/location/src/main.c

```diff
@@ -0,0 +1,297 @@
+/*
+ * Copyright (c) 2021 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#include <string.h>
+#include <zephyr.h>
+#include <nrf_modem_at.h>
+#include <modem/lte_lc.h>
+#include <modem/location.h>
+#if defined(CONFIG_NRF_CLOUD_PGPS)
+#include <date_time.h>
+#endif /* CONFIG_NRF_CLOUD_PGPS */
+
+static struct location_config config;
+static struct location_method_config methods[3];
+
+K_SEM_DEFINE(location_event, 0, 1);
+
+static void antenna_configure(void)
+{
+	int err;
+
+	if (strlen(CONFIG_LOCATION_SAMPLE_GNSS_AT_MAGPIO) > 0) {
+		err = nrf_modem_at_printf("%s", CONFIG_LOCATION_SAMPLE_GNSS_AT_MAGPIO);
+		if (err) {
+			printk("Failed to set MAGPIO configuration\n");
+		}
+	}
+
+	if (strlen(CONFIG_LOCATION_SAMPLE_GNSS_AT_COEX0) > 0) {
+		err = nrf_modem_at_printf("%s", CONFIG_LOCATION_SAMPLE_GNSS_AT_COEX0);
+		if (err) {
+			printk("Failed to set COEX0 configuration\n");
+		}
+	}
+}
+
+#if defined(CONFIG_NRF_CLOUD_PGPS)
+K_SEM_DEFINE(time_update_finished, 0, 1);
+
+static void date_time_evt_handler(const struct date_time_evt *evt)
+{
+	/* Don't care about the event, just indicate that we got it. */
+	k_sem_give(&time_update_finished);
+}
+
+static void current_time_get(void)
+{
+	int err;
+
+	/* If we wait for a while after LTE attach, it's more likely that modem has got the current
+	 * time from the LTE network and no NTP is needed.
+	 */
+	k_sleep(K_SECONDS(1));
+
+	err = date_time_update_async(date_time_evt_handler);
+	if (err) {
+		printk("Requesting the current time failed, error: %d\n", err);
+		return;
+	}
+
+	k_sem_take(&time_update_finished, K_MINUTES(1));
+}
+#endif /* CONFIG_NRF_CLOUD_PGPS */
+
+static const char *method_string_get(enum location_method method)
+{
+	switch (method) {
+	case LOCATION_METHOD_CELLULAR:
+		return "cellular";
+
+	case LOCATION_METHOD_GNSS:
+		return "GNSS";
+
+	case LOCATION_METHOD_WIFI:
+		return "Wi-Fi";
+
+	default:
+		return "unknown";
+	}
+}
+
+static void location_event_handler(const struct location_event_data *event_data)
+{
+	switch (event_data->id) {
+	case LOCATION_EVT_LOCATION:
+		printk("Got location:\n");
```

Comment:
Logger is enabled because it's used by the libraries. I think this makes it more clear what the sample is printing and what's coming from libraries used by the sample. I though about using logger also in the sample, but I think it's clearer this way. And there are other samples using `printk()` as well.

Reply:
@simensrostad Please check this and either resolve or add a further comment.